app/src/main/java/com/gophillygo/app/data/DestinationRepository.java

```diff
@@ -0,0 +1,84 @@
+package com.gophillygo.app.data;
+
+import android.arch.lifecycle.LiveData;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import com.gophillygo.app.data.models.Destination;
+import com.gophillygo.app.data.models.DestinationQueryResponse;
+import com.gophillygo.app.data.networkresource.ApiResponse;
+import com.gophillygo.app.data.networkresource.NetworkBoundResource;
+import com.gophillygo.app.data.networkresource.Resource;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+/**
+ * Mediator between Destination persistent data store and web service.
+ * Handles querying for data and loading it into the data store.
+ *
+ * Based on:
+ * https://developer.android.com/topic/libraries/architecture/guide.html
+ */
+
+class DestinationRepository {
+    private static final String LOG_LABEL = "DestinationRepository";
+
+    private final DestinationWebservice webservice;
+    private final DestinationDao dao;
+
+    // maximum rate at which to refresh data from network
+    private static final long RATE_LIMIT = TimeUnit.MINUTES.toMillis(15);
+
+    @Inject
+    public DestinationRepository(DestinationWebservice webservice,
+                                 DestinationDao dao) {
+        this.webservice = webservice;
+        this.dao = dao;
+    }
+
+    public LiveData<Destination> getDestination(String destinationId) {
+        // return a LiveData item directly from the database.
+        return dao.getDestination(destinationId);
+    }
+
+    public LiveData<Resource<List<Destination>>> loadDestinations() {
+        return new NetworkBoundResource<List<Destination>, DestinationQueryResponse>() {
+            @Override
+            protected void saveCallResult(@NonNull DestinationQueryResponse response) {
+                // clear out existing database entries before adding new ones
+                dao.clear();
+                for (Destination item: response.getDestinations()) {
+                    Log.d(LOG_LABEL, "item timestamp: " + item.getTimestamp());
+                    item.setTimestamp(System.currentTimeMillis());
```

Comment:
A better practice than using `currentTimeMillis()` for measuring an interval time would be to use either `uptimeMillis()` or `elapsedRealtime()`, as `currentTimeMillis()` represents wall-clock time and can "time travel" as the user changes time zones, changes their phone time, etc., whereas the latter two return a monotonic time.

See https://developer.android.com/reference/android/os/SystemClock.html for more details.

Reply:
Good information to reference!

Since persistent storage will survive system reboot and we may want to have long cache times (days), I think "wall" clock time is what we want, as apparent time-travel due to system reboot would be more frequent and harder to correct than clock time changes, which I think can be safely ignored for our caching purposes.